### ISSUE TITLE
Document how to enable debug logging in collector

### ISSUE THREAD

**Issue by ringerc** [open-telemetry/opentelemetry-operator#873]
The collector docs don't appear to mention how to control the collector's own log verbosity.

It uses Zap logging, and the k8s collector framework, so the admin may pass `--zap-log-level=debug` to set fine grained logging. This is documented for the operator SDK at https://sdk.operatorframework.io/docs/building-operators/golang/references/logging/

Mentioning the `--zap-log-level` param and linking to the SDK docs would help.

But it'd be nice to tell users how to actually set it, since Kustomize is a typical deployment method, but doing so through Kustomize is non-obvious. See https://github.com/operator-framework/operator-sdk/pull/5766/files for my patch against the operator SDK to add an explanation to its docs. A snippet like

```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
- https://url-of-opentelemetry-operator/
patches:
  - target:
      kind: Deployment
      name: opentelemetry-operator-controller-manager
      namespace: opentelemetry-opertor-system
    patch: |-
      - op: add
        path: "/spec/template/spec/containers/1/args/-"
        value: "--zap-log-level=debug"
```

or a `kubectl` hot-patch like

```
kubectl patch -n opentelemetry-operator-system $(\
  kubectl get -n opentelemetry-operator-system \
    -l app.kubernetes.io/name=opentelemetry-operator deployment -o name \
  ) \
  --type=json \
  -p '[{"op": "add", "path": "/spec/template/spec/containers/1/args/-", "value": "--zap-log-level=debug"}]' \
  --dry-run=client 
```

(To actually apply, remove `--dry-run=client` from the above)

TBH, it'd be a *lot* nicer to do this with an env-var binding, not a command-line argument. But documenting a simple way to adjust logging would help.

I don't see any increased logging with the above, so I'm not certain it's actually correct.

**Comment by pavolloffay** [comment]
@ringerc could you please comment?

**Comment by ringerc** [comment]
@pavolloffay  I'm suggesting that the docs for the operator cover how to raise the debug level of the operator. Managing the log level of the deployed collector is simple and already well documented.

It should IMO at least mention the `--zap-log-level=debug` option in the README, and link to the collector SDK docs for details.

But it'd be really nice to provide a quick explanation of how to enable it for the most common deployment methods of the operator, like I gave for Kustomize above.

IMO the operator SDK should really accept an env-var for the debug level. I'm looking at adding one. But in the mean time, helping users find out how to turn on debug logging for the operator would be helpful, as it took me a while to (a) find the options and (b) figure out how to apply them.

**Comment by pavolloffay** [comment]
thanks for the clarification. Would you mind opening an issue against the operator-sdk and linking it with this one? 

I totally agree that the log level (and other operator flags) should be configurable via env var.

**Comment by ringerc** [comment]
@pavolloffay I opened a PR already @ https://github.com/operator-framework/operator-sdk/pull/5766

**Comment by ringerc** [comment]
BTW @pavolloffay I'm thinking of attempting a much bigger change for the operator at https://github.com/open-telemetry/opentelemetry-operator/issues/901 that would address a large set of related open issues. But I really need some project owner input before I do anything, as it's a bit too large to attempt as a throw-away patch that might just get discarded.

I suspect the logging env-var should actually be pushed up to the operator-sdk rather than done in each operator. Think they're likely to be accepting of the idea? I could probably concoct a patch for it at the operator-sdk level if so.

**Comment by jaronoff97** [comment]
I think this was closed by #1193 let me know if that's not the case

**Comment by ringerc** [comment]
@jaronoff97 While the linked PR is indeed helpful it lacks the "document" part. Mention of that in the README would be helpful.
